### PR TITLE
Add Vertex Data blending notebook 

### PR DESCRIPTION
**Pull request by ShantamGupta**
Added vertexai search with data blending file

# Description

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [ ] You are listed as the author in your notebook or README file.
  - [ ] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [ ] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [ ] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by holtskinner — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have a few suggestions for readability/clarity, but otherwise good.

**Comment by holtskinner** [comment]
Also, if you wanted to go extra on this, you could add a section for how to do it with LangChain. (Although, I think the library would need to be updated to query based on `engine_id` instead of `datastore_id`)

**Comment by holtskinner** [comment]
Also, list yourself as the author in the top of the notebook. Similar to this in https://github.com/GoogleCloudPlatform/generative-ai/blob/main/gemini/getting-started/intro_gemini_python.ipynb
Screenshot 2024-03-14 at 10 08 38 AM

File: search/vertexai_search_data_blending.ipynb [hunk 1]
```diff
@@ -0,0 +1,396 @@
+{
```
[the remark this answers is not in the document]
Comment (ShantamGupta) [reply search/vertexai_search_data_blending.ipynb]: <p>done</p>

[the remark this answers is not in the document]
Comment (ShantamGupta) [reply search/vertexai_search_data_blending.ipynb]: <p>removed. done</p>

[the remark this answers is not in the document]
Comment (ShantamGupta) [reply search/vertexai_search_data_blending.ipynb]: <p>done</p>

[the remark this answers is not in the document]
Comment (ShantamGupta) [reply search/vertexai_search_data_blending.ipynb]: <p>removed. done</p>

[the remark this answers is not in the document]
Comment (ShantamGupta) [reply search/vertexai_search_data_blending.ipynb]: <p>done</p>

[the remark this answers is not in the document]
Comment (ShantamGupta) [reply search/vertexai_search_data_blending.ipynb]: <p>done. </p>

**Comment by ShantamGupta** [comment]
made the changes as per suggested edits. Thanks

File: search/vertexai_search_data_blending.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (holtskinner) [reply search/vertexai_search_data_blending.ipynb]: What do you mean the Python SDK isn't available yet? I have it working in my sample.

**Comment by holtskinner** [comment]
I have a couple of follow-ups. Specifically, the discoveryengine client library supports queries with Blended Search. The sample has been updated here https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/discoveryengine/search_sample.py

Can you update your code to use the client library?

**Comment by ShantamGupta** [comment]
So i already tried it a couple of days ago and retried the same code you shared. It returns the following issues 
InvalidArgument: 400 Setting query_expansion_spec, spell_correction_spec, content_search_spec proto fields is not allowed for multi-datastore search.

Which aligns with my discussion with PM and eng team. It doesn't support content spec and summarization yet..

**Review by holtskinner — APPROVED** [review]

**Pull request merged**